Tasks/Npm/npmtask.ts

```diff
@@ -1,27 +1,194 @@
+import Q = require('q');
+import fs = require('fs');
 import path = require('path');
+import url = require('url');
 import tl = require('vsts-task-lib/task');
+import trm = require('vsts-task-lib/toolrunner');
```

Comment:
There was some around getting access token and calculating the url with "pkgs" in it. I duplicated it deliberately since the nuget tasks/common depend on the old version of the lib.